cmd2.py

```diff
@@ -300,20 +299,32 @@ def path_complete(text, line, begidx, endidx, dir_exe_only=False, dir_only=False
     :return: List[str] - a list of possible tab completions
     """
 
+    # Get all tokens prior to token being completed
+    try:
+        prev_space_index = max(line.rfind(' ', 0, begidx), 0)
+        tokens = shlex.split(line[:prev_space_index], posix=POSIX_SHLEX)
+    except ValueError:
+        # Invalid syntax for shlex (Probably due to missing closing quote)
+        return []
+
+    if len(tokens) == 0:
+        return []
+
     # Determine if a trailing separator should be appended to directory completions
     add_trailing_sep_if_dir = False
     if endidx == len(line) or (endidx < len(line) and line[endidx] != os.path.sep):
         add_trailing_sep_if_dir = True
 
     add_sep_after_tilde = False
-    # If no path and no search text has been entered, then search in the CWD for *
-    if not text and line[begidx - 1] == ' ' and (begidx >= len(line) or line[begidx] == ' '):
+
+    # Readline places begidx after ~ and path separators (/) so we need to extract any directory
```

Comment:
Thank you for the improved comments throughout these tab-completion helper functions!  The behavior of ``readline`` is somewhat obscure, so these comments should help with maintainability of this code.